cpp/meta_module/algorithm/meta.cpp

```diff
@@ -0,0 +1,284 @@
+#include "meta.hpp"
+#include <cstdint>
+#include <string>
+#include <string_view>
+#include "mgp.hpp"
+
+namespace Meta {
+
+namespace {
+
+class Metadata {
+ public:
+  Metadata() { node_cnt = relationship_cnt = 0; }
+
+  int64_t node_cnt;
+  int64_t relationship_cnt;
+  std::unordered_map<std::string, int64_t> labels;
+  std::unordered_map<std::string, int64_t> property_key_cnt;
+  std::unordered_map<std::string, int64_t> relationship_types;
+  std::unordered_map<std::string, int64_t> relationship_types_cnt;
+
+  int64_t get_label_count() const { return static_cast<int64_t>(labels.size()); }
+  int64_t get_relationship_type_count() const { return static_cast<int64_t>(relationship_types_cnt.size()); }
+  int64_t get_property_key_count() const { return static_cast<int64_t>(property_key_cnt.size()); }
+
+  void reset();
+  void update_labels(const mgp::Node &node, int add);
+  void update_property_key_cnt(const mgp::Node &node, int add);
+  void update_property_key_cnt(const mgp::Relationship &relationship, int add);
+  void update_relationship_types(const mgp::Relationship &relationship, int add);
+  void update_relationship_types_cnt(const mgp::Relationship &relationship, int add);
+};
+
+// Global variable
+// This is unsafe in the multithreaded environment, the workaround would be building a thread-safe dynamic storage
+// implementation
+Metadata metadata;
+
+void insert(std::unordered_map<std::string, int64_t> &map, std::string &key, int64_t add) {
+  auto iterator = map.find(key);
+  if (iterator != map.end()) {
+    (*iterator).second += add;
+    if ((*iterator).second == 0) {
+      map.erase(iterator);
+    }
+  } else {
+    map[key] = add;
+  }
+}
+
+void insert(std::unordered_map<std::string, int64_t> &map, std::string_view key_view, int64_t add) {
+  auto key = std::string(key_view);
+  insert(map, key, add);
```

Comment:
```suggestion
  insert(map, std::move(key), add);
```

Reply:
or just remove this function, seems unnecessary. If you need std::string you can even cast it in the upper function